proxy/proxy.go

```diff
@@ -915,38 +920,7 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, *proxyError) {
 	p.metrics.IncCounter("outgoing." + req.Proto)
 	ctx.proxySpan.LogKV("http_roundtrip", StartEvent)
 
-	var response *http.Response
```

Comment:
Moving switch into `getRoundTripper` does not change anything for `http` but brings back clear roundtrip point as it was before fastcgi was introduced:
https://github.com/zalando/skipper/blob/a0c9feef7d932305c6f0f9406a362d6063e92605/proxy/proxy.go#L883-L887